src/Gui/propertyeditor/PropertyModel.cpp

```diff
@@ -572,6 +572,24 @@ void PropertyModel::removeProperty(const App::Property& _prop)
     }
 }
 
+void PropertyModel::renameProperty(const App::Property& _prop)
+{
+    auto prop = const_cast<App::Property*>(&_prop);
+    auto it = itemMap.find(prop);
+    if (it == itemMap.end() || !it->second) {
+        return;
+    }
+
+    PropertyItem* item = it->second;
+    item->renameProperty(prop);
+    QModelIndex parent = this->index(item->parent()->row(), 0, QModelIndex());
+    QModelIndex nameIndex = this->index(item->row(), 0, parent);
+    QModelIndex dataIndex = this->index(item->row(), 1, parent);
```

Comment:
Would be good to not rely on ints for indices here, but from what I learned from @pieterhijma it is typical in that file - so no reason to block that PR over it - a good opportunity for refactor, however!